### PR TITLE
WIP changes for purecap RISCV

### PR DESCRIPTION
**Pull request by arichardson**
Not ready to be merged, but some commits in this PR might be ready to be cherry-picked.

### CODE REVIEW

File: lib/csu/riscv/Makefile [hunk 1]
```diff
@@ -2,7 +2,11 @@
 
 .PATH: ${.CURDIR:H}/common
 
-SRCS=		crt1.c crti.S crtn.S
```
Comment (brooksdavis) [inline lib/csu/riscv/Makefile]: We almost certainly want a lib/csu/riscv64c like we do with mips64c128.

File: sys/riscv/include/_align.h [hunk 2]
```diff
@@ -38,7 +38,7 @@
  * for all data types (int, long, ...).   The result is unsigned int
  * and must be cast to any desired pointer type.
  */
-#define	_ALIGNBYTES	(sizeof(long long) - 1)
-#define	_ALIGN(p)	(((u_long)(p) + _ALIGNBYTES) & ~_ALIGNBYTES)
+#define	_ALIGNBYTES	(_Alignof(__max_align_t) - 1)
```
Comment (brooksdavis) [inline sys/riscv/include/_align.h]: I've pushed a different version using sizeof(void *) matching mips. I think that's what's meant here.

File: lib/libc/riscv/static_tls.h [hunk 3]
```diff
@@ -38,7 +38,12 @@ _libc_get_static_tls_base(size_t offset)
 {
 	uintptr_t tlsbase;
 
+#ifdef __CHERI_PURE_CAPABILITY__
+	__asm __volatile("cmove %0, ctp" : "=C"(tlsbase));
```
Comment (brooksdavis) [inline lib/libc/riscv/static_tls.h]: I've pushed something equivalent.

**Comment by arichardson** [comment]
I think all these changes (or better alternatives) have been committed.

**Pull request closed without merging**